src/plugins/intel_gpu/src/graph/broadcast.cpp

```diff
@@ -26,6 +28,31 @@ layout broadcast_inst::calc_output_layout(broadcast_node const& node) {
     return {input_layout.data_type, input_layout.format, desc->broadcast_sizes};
 }
 
+std::vector<layout> broadcast_inst::calc_output_layouts(broadcast_node const& node, const kernel_impl_params& impl_param) {
+    auto desc = node.get_primitive();
+    auto input_layout = node.get_dependency(0).get_output_layout();
+
+    ov::op::v3::Broadcast op;
+    op.set_broadcast_spec(ov::op::BroadcastType::NUMPY);
+
+    std::vector<ov::PartialShape> output_shapes = {ov::PartialShape()};
+    std::vector<ov::PartialShape> input_shapes = {
+        node.get_dependency(0).get_output_layout().get_partial_shape(),
+        ov::PartialShape{4},
+    };
+
+    std::vector<int64_t> target_shape = {1, 2, 10, 20};
+
+    auto tensor1 = make_host_tensor({ov::PartialShape{4}, data_types::i64, format::bfyx}, target_shape.data());
+    std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>> const_data = {
+        {1, tensor1},
+    };
+    ov::op::v3::shape_infer(&op, input_shapes, output_shapes, const_data);
+    auto output_format = format::get_default_format(output_shapes[0].size());
+
+    return { layout{output_shapes[0], input_layout.data_type, output_format} };
+}
```

Comment:
BTW, do we need to do shape_infer for broadcast? 
It will be either of the cases 
(1) static : Use the given output_shape set atop creation 
(2) If the output_shape is not given at the op creation 
- Read the input(1)'s output and use it as output shape 
Isnt this enough for this OP?

Reply:
I think the overall rule should be the following:
 - Easy shape infer functions can be implemented directly in the plugin w/o ngraph calls (e.g. for the cases when output_shape = input_shape)
 - Otherwise, ngraph should be called. 
 
Not sure about Broadcast, as it has additional parameter (axes set) which specifies the mapping between input and output dimensions which may require some additional logic for shape inference...